lib/index.js

```diff
@@ -70,11 +49,60 @@ Chartbeat.prototype.loaded = function() {
  */
 
 Chartbeat.prototype.page = function(page) {
+  this.updateConfig(page);
+
+  // since chartbeat automatically calls a page when it loads, don't load chartbeat script until 
+  // first Segment page call comes in and configures global config vars using its props
+  if (!this.pageCalledYet) {
+    this._ready = false;  // switch ready to false so that no pages after the first one can fire until _initialize has loaded chartbeat script
+    this.pageCalledYet = true;
+    this._initialize();
+  } else {
+    var props = page.properties();
+    window.pSUPERFLY.virtualPage(props.path);
+  }
+};
+
+// update chartbeat global config vars
+Chartbeat.prototype.updateConfig = function(page) {
   var category = page.category();
-  if (category) window._sf_async_config.sections = category;
   var author = page.proxy('properties.author');
-  if (author) window._sf_async_config.authors = author;
   var props = page.properties();
-  var name = page.fullName();
-  window.pSUPERFLY.virtualPage(props.path, name || props.title);
+  var title = page.fullName() || props.title;
+
+  // update general config
+  window._sf_async_config = window._sf_async_config || {};
+
+  if (category) window._sf_async_config.sections = category;
+  if (author) window._sf_async_config.authors = author;
+  if (title) window._sf_async_config.title = title;
+
+  // update subscriber engagement
+  var _cbq = window._cbq = window._cbq || [];
+
+  for (var key in props) {
+    if (!props.hasOwnProperty(key)) continue;
+    if (this.options.subscriberEngagementKeys.indexOf(key) > -1) {
+      _cbq.push([key, props[key]]);
+    }
+  }
 };
+
+// sets global vars and loads Chartbeat script
+Chartbeat.prototype._initialize = function() {
+  var self = this;
+  var script = this.options.video ? 'chartbeat_video.js' : 'chartbeat.js';
+
+  window._sf_async_config.useCanonical = true;
+  defaults(window._sf_async_config, {
+    domain: this.options.domain,
+    uid: this.options.uid
+  });
+
+  onBody(function() {
+    window._sf_endpt = new Date().getTime();
+    // Note: Chartbeat depends on document.body existing so the script does
+    // not load until that is confirmed. Otherwise it may trigger errors.
+    self.load({ script: script }, self.ready);  // switch ready to true for real once the script has loaded
```

Comment:
also wonder about setting ready here as well. just so I understand, this sets `ready = true`? when it had been disabled after the first page call?

Reply:
Correct.

Yeah so what'll happen is:
1. it will emit 'ready' https://github.com/segmentio/analytics.js-core/blob/802a41295035e0ff80e5ef059e105d39fb9bd544/lib/analytics.js#L133-L136

2. A listener will call flush https://github.com/segmentio/analytics.js-integration/blob/master/lib/index.js#L40

3. `flush` will set `_ready` to true https://github.com/segmentio/analytics.js-integration/blob/master/lib/protos.js#L208

Below is the code that causes `page` calls to be queued based on the value of `_ready`

https://github.com/segmentio/analytics.js-core/blob/802a41295035e0ff80e5ef059e105d39fb9bd544/lib/analytics.js#L466

https://github.com/segmentio/analytics.js-integration/blob/d576c245a07782c1590bcbc688d8ce9a22be4f08/lib/protos.js#L176